### ISSUE TITLE
对DeepSeek-VL2进行GRPO训练

### ISSUE THREAD

**Issue by xdaiycl** [modelscope/ms-swift#4775]
对DeepSeek-VL2进行GRPO训练，transformers版本为4.41.2，报错为要求更新trl，但更新trl会导致transformers更新，进而导致不支持deepseek-vl，因为它要求是transformers<4.42，这就比较尴尬

**Comment by hjh0119** [comment]
Maybe you can ignore the version incompatibility warnings. Are there any errors during the training process?

**Comment by xdaiycl** [comment]
> Maybe you can ignore the version incompatibility warnings. Are there any errors during the training process?

有的，是这个错误，https://github.com/modelscope/ms-swift/issues/3338

**Comment by hjh0119** [comment]
I mean, did you encounter any errors when training with Transformers 4.41?

**Comment by xdaiycl** [comment]
> I mean, did you encounter any errors when training with Transformers 4.41?

有的，在使用transformers4.41.2时，SFT可以跑通；但在GRPO时会报错ImportError: cannot import name 'GRPOConfig' from 'trl'；此时，trl为0.12.2；我记得使用GRPO要求trl>0.17.0，更新trl又会与transformers不兼容

**Comment by hjh0119** [comment]
transformers 4.41.2
trl 0.18

Ignore environment incompatibility warnings.

**Comment by xdaiycl** [comment]
仍旧报错AttributeError: 'DeepseekV2ForCausalLM' object has no attribute 'generate'；Shell如下：CUDA_VISIBLE_DEVICES=0,1 \
NPROC_PER_NODE=2 \
swift rlhf \
  --rlhf_type grpo \
  --model deepseek-ai/deepseek-vl2-tiny \
  --train_type lora \
  --use_vllm true \
  --vllm_mode colocate \
  --vllm_gpu_memory_utilization 0.5 \
  --vllm_max_model_len 8192 \
  --vllm_tensor_parallel_size 2 \
  --dataset /data/FinTabNet_train_GRPO.jsonl \
  --val_dataset /data//FinTabNet_val_GRPO.jsonl \
  --external_plugins examples/train/grpo/plugin/plugin.py \
  --reward_funcs external_teds external_teds_struct format \
  --reward_weights 1 1 0.1 \
  --torch_dtype bfloat16 \
  --num_train_epochs 1 \
  --max_length 8192 \
  --per_device_train_batch_size 8 \
  --per_device_eval_batch_size 8 \
  --gradient_accumulation_steps 1 \
  --eval_steps 10000 \
  --save_steps 10000 \
  --learning_rate 1e-6 \
  --save_total_limit 2 \
  --logging_steps 1 \
  --warmup_ratio 0.05 \
  --dataloader_num_workers 4 \
  --max_completion_length 2048 \
  --num_generations 8 \
  --deepspeed zero3_offload \
  --temperature 1.1 \
  --top_p 1.0 \
  --top_k 80 \
  --log_completions true \
  --async_generate false \
  --offload_optimizer true \
  --offload_model true \
  --gc_collect_after_offload true \
  --move_model_batches 40 \
  --sleep_level 1 \
  --report_to wandb \
  --system examples/train/grpo/prompt.txt；环境如下：Package                                  Version       Editable project location
---------------------------------------- ------------- ----------------------------------------------------------------------------------------------------------
absl-py                                  2.2.2
accelerate                               1.6.0
addict                                   2.4.0
aiofiles                                 24.1.0
aiohappyeyeballs                         2.6.1
aiohttp                                  3.11.16
aiohttp-cors                             0.8.1
aiosignal                                1.3.2
airportsdata                             20250224
aliyun-python-sdk-core                   2.16.0
aliyun-python-sdk-kms                    2.16.5
annotated-types                          0.7.0
antlr4-python3-runtime                   4.13.2
anyio                                    4.9.0
apted                                    1.0.3
astor                                    0.8.1
asttokens                                3.0.0
async-timeout                            5.0.1
attrdict                                 2.0.1
attrs                                    25.3.0
beautifulsoup4                           4.12.3
binpacking                               1.5.2
blake3                                   1.0.4
cachetools                               5.5.2
certifi                                  2025.1.31
cffi                                     1.17.1
charset-normalizer                       3.4.2
click                                    8.1.8
cloudpickle                              3.1.1
cmake                                    4.0.0
cobble                                   0.1.4
colorama                                 0.4.6
colorful                                 0.5.7
compressed-tensors                       0.9.0
contourpy                                1.3.1
cpm-kernels                              1.0.11
crcmod                                   1.7
cryptography                             44.0.2
cupy-cuda12x                             13.4.1
cycler                                   0.12.1
dacite                                   1.9.2
datasets                                 3.2.0
deepspeed                                0.15.4
defusedxml                               0.7.1
Deprecated                               1.2.18
depyf                                    0.18.0
dill                                     0.3.8
diskcache                                5.6.3
Distance                                 0.1.3
distlib                                  0.3.9
distro                                   1.9.0
dnspython                                2.7.0
einops                                   0.8.1
email_validator                          2.2.0
et_xmlfile                               2.0.0
exceptiongroup                           1.2.2
executing                                2.2.0
fastapi                                  0.115.12
fastapi-cli                              0.0.7
fastrlock                                0.8.3
ffmpy                                    0.5.0
filelock                                 3.18.0
fonttools                                4.57.0
frozenlist                               1.5.0
fsspec                                   2024.9.0
future                                   1.0.0
gguf                                     0.10.0
google-api-core                          2.25.1
google-auth                              2.40.3
googleapis-common-protos                 1.70.0
gradio                                   5.25.0
gradio_client                            1.8.0
groovy                                   0.1.2
grpcio                                   1.71.0
h11                                      0.14.0
hf-xet                                   1.1.0
hjson                                    3.1.0
httpcore                                 1.0.8
httptools                                0.6.4
httpx                                    0.28.1
huggingface-hub                          0.30.2
icecream                                 2.1.5
idna                                     3.10
importlib_metadata                       8.0.0
interegular                              0.3.3
jieba                                    0.42.1
Jinja2                                   3.1.6
jiter                                    0.9.0
jmespath                                 0.10.0
joblib                                   1.4.2
jsonschema                               4.23.0
jsonschema-specifications                2025.4.1
kiwisolver                               1.4.8
lark                                     1.2.2
latex2sympy2_extended                    1.10.1
Levenshtein                              0.27.1
llguidance                               0.7.19
llvmlite                                 0.44.0
lm-format-enforcer                       0.10.1
lxml                                     5.3.0
mammoth                                  1.8.0
Markdown                                 3.8
markdown-it-py                           3.0.0
markdownify                              0.14.1
MarkupSafe                               3.0.2
math-verify                              0.7.0
matplotlib                               3.10.1
mdurl                                    0.1.2
mistral_common                           1.5.4
modelscope                               1.25.0
mpmath                                   1.3.0
ms_swift                                 3.6.0.dev0    /data/ms-swift-main
msgpack                                  1.1.0
msgspec                                  0.19.0
multidict                                6.4.3
multiprocess                             0.70.16
nest-asyncio                             1.6.0
networkx                                 3.4.2
ninja                                    1.11.1.4
nltk                                     3.9.1
numba                                    0.61.2
numpy                                    1.26.4
nvidia-cublas-cu12                       12.1.3.1
nvidia-cuda-cupti-cu12                   12.1.105
nvidia-cuda-nvrtc-cu12                   12.1.105
nvidia-cuda-runtime-cu12                 12.1.105
nvidia-cudnn-cu12                        8.9.2.26
nvidia-cufft-cu12                        11.0.2.54
nvidia-curand-cu12                       10.3.2.106
nvidia-cusolver-cu12                     11.4.5.107
nvidia-cusparse-cu12                     12.1.0.106
nvidia-cusparselt-cu12                   0.6.2
nvidia-ml-py                             12.570.86
nvidia-nccl-cu12                         2.20.5
nvidia-nvjitlink-cu12                    12.4.127
nvidia-nvtx-cu12                         12.1.105
openai                                   1.73.0
opencensus                               0.11.4
opencensus-context                       0.1.3
opencv-python-headless                   4.11.0.86
openpyxl                                 3.1.5
opentelemetry-api                        1.26.0
opentelemetry-exporter-otlp              1.26.0
opentelemetry-exporter-otlp-proto-common 1.26.0
opentelemetry-exporter-otlp-proto-grpc   1.26.0
opentelemetry-exporter-otlp-proto-http   1.26.0
opentelemetry-proto                      1.26.0
opentelemetry-sdk                        1.26.0
opentelemetry-semantic-conventions       0.47b0
opentelemetry-semantic-conventions-ai    0.4.5
orjson                                   3.10.16
oss2                                     2.19.1
outlines                                 0.1.11
outlines_core                            0.1.26
packaging                                24.2
pandas                                   2.2.3
partial-json-parser                      0.2.1.1.post5
pathvalidate                             3.2.1
peft                                     0.13.0
pillow                                   11.2.1
pip                                      25.0
platformdirs                             4.3.8
prometheus_client                        0.21.1
prometheus-fastapi-instrumentator        7.1.0
propcache                                0.3.1
proto-plus                               1.26.1
protobuf                                 4.25.7
psutil                                   7.0.0
puremagic                                1.28
py-cpuinfo                               9.0.0
py-spy                                   0.4.0
pyarrow                                  19.0.1
pyasn1                                   0.6.1
pyasn1_modules                           0.4.2
pycountry                                24.6.1
pycparser                                2.22
pycryptodome                             3.22.0
pydantic                                 2.11.3
pydantic_core                            2.33.1
pydub                                    0.25.1
Pygments                                 2.19.1
pyparsing                                3.2.3
python-dateutil                          2.9.0.post0
python-dotenv                            1.1.0
python-json-logger                       3.3.0
python-multipart                         0.0.20
python-pptx                              1.0.2
pytz                                     2025.2
PyYAML                                   6.0.2
pyzmq                                    26.4.0
RapidFuzz                                3.13.0
ray                                      2.45.0
referencing                              0.36.2
regex                                    2024.11.6
requests                                 2.32.3
rich                                     14.0.0
rich-toolkit                             0.14.4
rouge                                    1.0.1
rpds-py                                  0.24.0
rsa                                      4.9.1
ruff                                     0.11.5
safehttpx                                0.1.6
safetensors                              0.5.3
scipy                                    1.15.2
semantic-version                         2.10.0
sentencepiece                            0.2.0
setuptools                               69.5.1
shellingham                              1.5.4
simplejson                               3.20.1
six                                      1.17.0
smart-open                               7.1.0
sniffio                                  1.3.1
sortedcontainers                         2.4.0
soupsieve                                2.6
SpeechRecognition                        3.12.0
starlette                                0.46.1
sympy                                    1.13.1
tensorboard                              2.19.0
tensorboard-data-server                  0.7.2
tiktoken                                 0.9.0
timm                                     1.0.15
tokenizers                               0.21.2
tomlkit                                  0.13.2
torch                                    2.3.0
torchaudio                               2.5.1
torchvision                              0.18.0
tqdm                                     4.67.1
transformers                             4.53.0
transformers-stream-generator            0.0.5
triton                                   2.3.0
trl                                      0.17.0
typer                                    0.15.2
typing_extensions                        4.13.2
typing-inspection                        0.4.0
tzdata                                   2024.2
urllib3                                  2.4.0
uvicorn                                  0.34.0
uvloop                                   0.21.0
virtualenv                               20.31.2
vllm                                     0.5.1
vllm-flash-attn                          2.5.9
watchfiles                               1.0.5
websockets                               15.0.1
Werkzeug                                 3.1.3
wheel                                    0.45.1
wrapt                                    1.17.2
xformers                                 0.0.26.post1
xgrammar                                 0.1.18
XlsxWriter                               3.2.0
xxhash                                   3.5.0
yarl                                     1.19.0
youtube-transcript-api                   0.6.3
zipp                                     3.21.0
zstandard                                0.23.0
